django_tests/test_auth_backend.py

```diff
@@ -20,90 +24,161 @@ def profile_medewerker():
+            "datasets": {"brk": {
+                "tables": {"kadastraleobjecten": {
+                    "fields": {"volgnummer": "encoded", "identificatie": "encoded"}}}}
+            }
         },
     )
 
 
 @pytest.fixture
-def profile_brk_readall():
+def profile_brk_read_full():
     return Profile.objects.create(
-        name="brk_readall",
+        name="brk_read_full",
         scopes="['BRK/RO','BRK/RSN']",
         schema_data={
             "datasets": {
-                "brk": {"tables": {"geheim": {"fields": {"id": "read", "bsn": "read"}}}}
+                "brk": {"tables": {"kadastraleobjecten": {"fields": {"id": "read", "volgnummer": "read"}}}}
             }
         },
     )
 
 
+@pytest.fixture()
+def brk_schema_json() -> dict:
+    path = TEST_FILES_FOLDER / "kadastraleobjecten.json"
+    return json.loads(path.read_text())
+
+
+@pytest.fixture()
+def brk_dataset(brk_schema_json) -> Dataset:
+    return Dataset.objects.create(name="brk", schema_data=brk_schema_json)
+
+
 @pytest.mark.django_db
 def test_get_profiles_for_request(profile_medewerker, profile_brk_read):
     """Check that correct Profiles returned for request."""
     request = RequestFactory().get("/")
-    request.is_authorized_for = lambda scopes: "FP/MD" in scopes
-
+    def auth(*scopes):
+        for scope in scopes:
+            if scope in ["FP/MD"]:
+                return True
+        return False
+    request.is_authorized_for = auth
     request_profile = RequestProfile(request)
-
     assert request_profile.get_profiles() == {profile_medewerker}
 
 
 @pytest.mark.django_db
-def test_has_perm_dataset(profile_medewerker, profile_brk_read):
-    """Check that BRK user can access data, while regular medeweker not."""
+def test_has_perm_dataset(profile_medewerker, profile_brk_read, brk_dataset):
+    """Check that user who is authorized for all authorization checks can access data"""
     request = RequestFactory().get("/")
-    request.is_authorized_for = lambda scopes: True  # Both Profiles used
 
+    def auth(*scopes):
+        return True
+    request.is_authorized_for = auth
     backend = ProfileAuthorizationBackend()
-
     user_1 = mock.MagicMock()
     user_1.request = request
-
-    assert backend.has_perm(user_1, "brk") is True
+    assert backend.has_perm(user_1, "brk:kadastraleobjecten:id") is True
 
 
 @pytest.mark.django_db
-def test_has_perm_dataset_table(profile_medewerker, profile_brk_read):
-    """Check that BRK user can access data, while regular medeweker not."""
+def test_has_perm_dataset_field_two_profiles(profile_medewerker, profile_brk_read, profile_brk_read_full, brk_dataset):
+    """Checks that profiles combine for a given user.
+    And that he can access data accordingly, while regular medeweker not."""
     request = RequestFactory().get("/")
-    request.is_authorized_for = lambda scopes: True  # Both Profiles used
+
+    def auth(*scopes):  # Both Profiles used
```

Comment:
Moet hier gewoon containment gechecked worden, dus of "BRK/RO" een van de scopes is? Dat zou mooi met een set kunnen, iets als `"BRK/RO" in set(scopes)`.

Reply:
Goed punt, is aangepast